src/VertexArray.cc

```diff
@@ -0,0 +1,488 @@
+#include <iostream>
+#include <iomanip>
+#include <sstream>
+#include <vector>
+#include <memory>
+#include <cstdio>
+#include <functional>
+#include "enums.h"
+#include "VertexArray.h"
+
+#include "printutils.h"
+
+void addAttributeValues(IAttributeData &) {}
+
+void VertexData::fillInterleavedBuffer(GLbyte* interleaved_buffer) const
+{
+	// All attribute vectors need to be the same size to interleave
+	if (attributes_.size()) {
+		size_t idx = 0, last_size = attributes_[0]->size() / attributes_[0]->count(), stride = stride_;
+#if 0
+	for (const auto &data : attributes_) {
+		if (idx != 0) {
+			if (last_size != data->size() / data->count()) {
+				PRINTDB("attribute data for vertex incorrect size at index %d = %d", idx % (data->size() / data->count()));
+				PRINTDB("last_size = %d", last_size);
+				assert(false);
+			}
+		}
+		last_size = data->size() / data->count();
+		idx++;
+		//stride += data->sizeofAttribute();
+	}
+#endif // 0
```

Comment:
I believe so. I changed some of the handling when I refactored createVertex into VertexArray, and I don't believe it can be created in this state, so the check was removed for speed.